core/src/main/java/org/apache/calcite/rel/core/JoinRelType.java

```diff
@@ -152,7 +153,7 @@ public JoinRelType cancelNullsOnRight() {
   }
 
   /** Transform this JoinRelType to CorrelateJoinType. **/
-  public CorrelateJoinType toLinq4j() {
+  public CorrelateJoinType toLinq4jCorrelateJoinType() {
```

Comment:
I would consider JoinRelType (and all public methods) as part of our public API. So we could say that this is a breaking change. On the other hand, I don't expect this method to be used by the clients so I wouldn't put it here in the first place. As far as I can see it is used only inside EnumerableCorrelate, how about moving the code there?

Reply:
If we decide to deprecate `CorrelateJoinType`, I'd rather keep this method unchanged for the moment, and mark it as deprecated at the same time that we deprecate `CorrelateJoinType`.